tests/baselines/reference/esDecorators-classDeclaration-commentPreservation(module=commonjs,target=es2015).js

```diff
@@ -208,7 +208,7 @@ let C = (() => {
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.D = void 0;
 /*34*/
-exports.D = (() => {
+let D = exports.D = (() => {

```

Comment:
If this uses a let, will code like:

```ts
class Foo {}
namespace Foo {
    export test = 1234;
}
```

Work correctly?

Reply:
It should. The merging behavior occurs in the `ts` transform, before decorators are processed.

Reply:
Is there a test we could add to verify?

Reply:
That behavior is already covered by tests/cases/conformance/internalModules/DeclarationMerging/ClassAndModuleThatMerge*.ts. Nothing I changed here would have any impact on declaration merging since that happens long before this transform runs.

Reply:
Right, I get that class/namespace merging is covered, but skimming the tests you link, none of them export and use CJS, so it doesn't seem like this particular bug + namespace merging is covered.

Am I missing a test that would exhibit both behaviors? I would assume it doesn't exist, otherwise the original bug would have also been caught?

Reply:
I don't think class/namespace merging has anything to do with the original bug, but I've added a test anyways.